vpc-endpoints.tf

```diff
@@ -0,0 +1,35 @@
+resource "aws_security_group" "sgforendpoint" {
+  name        = "EndpointSG"
+  description = "Allow indbound and outbound traffic for VPC endpoint"
+  vpc_id      = "${aws_vpc.main.id}"
+
+  ingress {
```

Comment:
Would recommend using `aws_security_group_rule` rather than inline blocks. The blocks are not perfectly idempotent when you're making a change to an existing resource.